ui/src/dropdown.tsx

```diff
@@ -246,11 +248,22 @@ const
         </Fluent.Dialog>
       </>
     )
+  },
+  getCompBasedOnDialogProp = (m: Dropdown) => {
+    m.dialog = m.dialog ?? 'auto'
+    switch (m.dialog) {
+      case 'always':
+        return <DialogDropdown {...m} />
+      case 'never':
+        return <BaseDropdown {...m} />
+      case 'auto':
+        return (m.choices?.length || 0) > 100 ? <DialogDropdown {...m} /> : <BaseDropdown {...m} />
+    }
   }
 
 export const XDropdown = ({ model: m }: { model: Dropdown }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   React.useEffect(() => { wave.args[m.name] = m.values ? (m.values || []) : (m.value || null) }, [])
-
-  return (m.choices?.length || 0) > 100 ? <DialogDropdown {...m} /> : <BaseDropdown {...m} />
+  
+  return getCompBasedOnDialogProp(m)
```

Comment:
I would just inline this like before instead of a separate function.

```
return m.dialog === 'always' 
    ? <DialogDropdown {...m} /> 
    : m.dialog === 'never' 
        ? <BaseDropdown {...m} /> 
        : (m.choices?.length || 0) > 100 
            ? <DialogDropdown {...m} /> 
            : <BaseDropdown {...m} />
```

Reply:
I like it. Inlined.